CMakeLists.txt

```diff
@@ -23,7 +23,21 @@ if(APPLE)
   # Looking at cmake source code OS_RELEASE is already set to the output of `sw_vers -productVersion` which is what we want
   cmake_host_system_information(RESULT OSX_VERSION QUERY OS_RELEASE)
   message("-- OS_RELEASE variable is set to: " ${OSX_VERSION})
+
+  if(DEFINED XCODE_VERSION)
+    if(XCODE_VERSION VERSION_LESS "13.0")
+      message(FATAL_ERROR "The minimum required version of XCode for compiling OpenStudio is 13, however XCODE_VERSION is set to " ${XCODE_VERSION} ". Please upgrade XCode and try again.")
+    endif()
+  # When using a generator other than XCode, ${XCODE_VERSION} is not set. Verify that we have at least the minimum macOS version required for XCode 13
+  elseif(OSX_VERSION VERSION_LESS "11.3")
+    message(FATAL_ERROR "The minimum required version of XCode for compiling OpenStudio is 13, which requires macOS 11.3 or greater, however macOS version is set to " ${OSX_VERSION} ". Please upgrade macOS and XCode and try again.")
+  endif()
+
   if(NOT CMAKE_OSX_DEPLOYMENT_TARGET STREQUAL "")
+    if(CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS "11.0")
```

Comment:
Thanks for this PR @mduvanel, where did you see the minimum deployment target being 11?  Currently we have it at 10.15 for Mac x86

https://github.com/openstudiocoalition/OpenStudioApplication/blob/develop/.github/workflows/app_build.yml#L97

Reply:
Oh I see, and it caused [this build](https://github.com/openstudiocoalition/OpenStudioApplication/actions/runs/7757378494/job/21202606913) to fail.

I based myself on this table from the Qt 6 documentation: https://doc.qt.io/qt-6/macos.html#supported-versions. Apparently the support for macOS 10.14 and 10.15 got dropped between [6.4](https://doc.qt.io/archives/qt-6.4/macos.html) and [6.5](https://doc.qt.io/archives/qt-6.5/macos.html).

Since the Qt version is set to 6.6.1 [here](https://github.com/openstudiocoalition/OpenStudioApplication/blob/4652e63f2aacc5ae75e41922bcc011701074b87d/CMakeLists.txt#L496) it may be safer to enforce this condition, especially based on this note from the same page:

```
Note: You should not lower the deployment target beyond the default value set by Qt. Doing so will likely lead to crashes at runtime if the binary is then deployed to a macOS version lower than what Qt expected to run on.
```

Reply:
I have added an extra commit that bumps `MACOSX_DEPLOYMENT_TARGET` to 11.0 in the CI scripts

Reply:
That documentation is pretty clear, seems like it is good to update.  I will update the release notes too, thanks!